root/static/scripts/tests/edit/utility/getUpdatedTrackArtists.js

```diff
@@ -0,0 +1,133 @@
+    },
+    {
+      desc: 'artist change, track doesn\'t match old',
+      track: [name('Other', '3')],
+      oldRel: [name('Old', '1')],
+      newRel: [name('New', '2')],
+      want: [name('Other', '3')],
+    },
+    {
+      desc: 'added credited-as, track matches old',
+      track: [name('Old', '1')],
+      oldRel: [name('Old', '1')],
+      newRel: [name('Old', '1', '', 'Alt')],
+      want: [name('Old', '1', '', 'Alt')],
+    },
+    {
+      desc: 'removed credited-as, track matches old',
+      track: [name('Old', '1', '', 'Alt')],
+      oldRel: [name('Old', '1', '', 'Alt')],
+      newRel: [name('Old', '1')],
+      want: [name('Old', '1')],
+    },
+    {
+      desc: 'multiple artists, track matches old',
+      track: [name('A', '1', ' & '), name('B', '2')],
+      oldRel: [name('A', '1', ' & '), name('B', '2')],
+      newRel: [name('C', '3', ' feat. '), name('D', '4')],
+      want: [name('C', '3', ' feat. '), name('D', '4')],
+    },
+    {
+      desc: 'multiple artists, track matches old plus feat.',
+      track: [
+        name('A', '1', ' & '),
+        name('B', '2', ' feat. '),
+        name('E', '5'),
+      ],
+      oldRel: [name('A', '1', ' & '), name('B', '2')],
+      newRel: [name('C', '3', ' with '), name('D', '4')],
+      want: [
+        name('C', '3', ' with '),
+        name('D', '4', ' feat. '),
+        name('E', '5'),
+      ],
+    },
+    {
+      desc: 'multiple artists, track matches first old',
+      track: [name('A', '1')],
+      oldRel: [name('A', '1', ' & '), name('B', '2')],
+      newRel: [name('C', '3', ' & '), name('D', '4')],
+      want: [name('A', '1')],
+    },
+    {
+      desc: 'multiple artists, track matches second old',
+      track: [name('B', '2')],
+      oldRel: [name('A', '1', ' & '), name('B', '2')],
+      newRel: [name('C', '3', ' & '), name('D', '4')],
+      want: [name('B', '2')],
+    },
+    {
+      desc: 'track matches new with diff. join phrase',
+      track: [name('C', '3', ' feat. '), name('D', '4')],
+      oldRel: [name('A', '1', ' & '), name('B', '2')],
+      newRel: [name('C', '3', ' & '), name('D', '4')],
+      want: [name('C', '3', ' & '), name('D', '4')],
+    },
+    {
+      desc: 'multiple artists with credited-as, track matches old',
+      track: [name('A', '1', ' & ', 'a'), name('B', '2')],
+      oldRel: [name('A', '1', ' & ', 'a'), name('B', '2')],
+      newRel: [name('A', '1', ' & '), name('B', '2')],
+      want: [name('A', '1', ' & '), name('B', '2')],
+    },
+    {
+      desc: 'multiple artists with credited-as, track has feat.',
+      track: [
+        name('A', '1', ' & ', 'a'),
+        name('B', '2', ' feat. '),
+        name('C', '3'),
+      ],
+      oldRel: [name('A', '1', ' & ', 'a'), name('B', '2')],
+      newRel: [name('A', '1', ' & '), name('B', '2')],
+      want: [
+        /*
+         * TODO: As discussed in MBS-13273, the first artist's credited-as
+         * name should be removed here.
+         */
```

Comment:
This is something that I plan to change, but I figured it'd be good to add a test case now to document the existing behavior.